policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceHierarchyProxy.java

```diff
@@ -42,101 +41,174 @@
+            LOGGER.debug("resource {} NOT an instance of ChildResource (top of hierarchy)", resource);
+            inheritedRules = Optional.empty();
         }
-        mergedRules.message(String.join(",", messages));
-        LOGGER.debug("Merged messages: {} + {} -> {}", inheritedRules.getMessage(), newRules.getMessage(), mergedRules.getMessage());
 
-        // Merge rules
-        mergedRules.addRules(inheritedRules.getRules());
-        mergedRules.addRules(newRules.getRules());
-        LOGGER.debug("Merged rules: {} + {} -> {}", inheritedRules.getRules(), newRules.getRules(), mergedRules.getRules());
+        Optional<Rules<LeafResource>> newRules = rulesExtractor.apply(resource);
 
-        return mergedRules;
+        // If both present, merge both
+        // If either present, return present
+        // If none present, return Optional.empty()
+        return inheritedRules.map(iRules -> newRules.map(nRules -> mergeRules(iRules, nRules)).or(() -> inheritedRules)).orElse(newRules);
+    }
+
+    /**
+     * GetRecordRules is used by the service to get any record rules that could be applied against the resource that the user has requested
+     * If no rules are found then a {@link NoSuchPolicyException} will be thrown
+     *
+     * @param resource a {@link LeafResource} to get rules for
+     * @return the record rules that apply to the LeafResource
+     */
+    public Rules<Serializable> getRecordRules(final LeafResource resource) {
+        var optionalRules = getRecordRules(resource, service::getRecordRules);
+
+        return optionalRules
+                .filter(rules -> !rules.getRules().isEmpty())
+                .orElseThrow(() -> new NoSuchPolicyException("No Policy Found"));
     }
 
-    @Override
-    public Optional<Policy> getPolicy(final Resource resource) {
-        Optional<Rules<Serializable>> optionalRules = getHierarchicalRules(resource, Policy::getRecordRules);
-        return optionalRules.map(rules -> new Policy<>().recordRules(rules));
+    /**
+     * GetResourceRules is used by the service to get any resource rules that could be applied against the resource.
+     * If no rules are applied then a {@link NoSuchPolicyException} will be thrown
+     * A resource rule may be applied at any point in the file tree, and could cause the record to be redacted.
+     *
+     * @param resource {@link Resource} the user wants access to, this could be a Directory, stream, system resource or file
+     * @return rules {@link Rules} object, which contains the list of rules found that need to be applied to the resource
+     */
+    public Rules<LeafResource> getResourceRules(final LeafResource resource) {
+        Optional<Rules<LeafResource>> optionalRules = getResourceRules(resource, service::getResourceRules);
+
+        return optionalRules
+                .filter(rules -> !rules.getRules().isEmpty())
+                .orElseThrow(() -> new NoSuchPolicyException("No Policy Found"));
```

Comment:
Ditto, no *resource* rules for this leaf resource

Reply:
updated